contracts/modules/grouping/GroupingModule.sol

```diff
@@ -0,0 +1,224 @@
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IGroupNFT public immutable GROUP_NFT;
+
+    /// @notice Returns the canonical protocol-wide Group IP Asset Registry
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IGroupIPAssetRegistry public immutable GROUP_IP_ASSET_REGISTRY;
+
+    /// @notice Returns the canonical protocol-wide LicenseRegistry
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ILicenseRegistry public immutable LICENSE_REGISTRY;
+
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.GroupingModule")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant GroupingModuleStorageLocation =
+        0x4f35861babcda7cb8a75afddcc0971d8dc0cbbd9d19afddbe94e0dcd72824100;
+
+    /// Constructor
+    /// @param accessController The address of the AccessController contract
+    /// @param ipAssetRegistry The address of the IpAssetRegistry contract
+    /// @param licenseToken The address of the LicenseToken contract
+    /// @param licenseRegistry The address of the LicenseRegistry contract
+    /// @param groupNFT The address of the GroupNFT contract
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(
+        address accessController,
+        address ipAssetRegistry,
+        address licenseRegistry,
+        address licenseToken,
+        address groupNFT
+    ) AccessControlled(accessController, ipAssetRegistry) {
+        if (licenseToken == address(0)) revert Errors.GroupingModule__ZeroLicenseToken();
+        if (licenseRegistry == address(0)) revert Errors.GroupingModule__ZeroLicenseRegistry();
+        if (groupNFT == address(0)) revert Errors.GroupingModule__ZeroGroupNFT();
+        if (ipAssetRegistry == address(0)) revert Errors.GroupingModule__ZeroIpAssetRegistry();
+
+        LICENSE_TOKEN = ILicenseToken(licenseToken);
+        GROUP_IP_ASSET_REGISTRY = IGroupIPAssetRegistry(ipAssetRegistry);
+        LICENSE_REGISTRY = ILicenseRegistry(licenseRegistry);
+
+        if (!groupNFT.supportsInterface(type(IGroupNFT).interfaceId)) {
+            revert Errors.GroupingModule__InvalidGroupNFT(groupNFT);
+        }
+        GROUP_NFT = IGroupNFT(groupNFT);
+        _disableInitializers();
+    }
+
+    /// @notice initializer for this implementation contract
+    /// @param accessManager The address of the protocol admin roles contract
+    function initialize(address accessManager) public initializer {
+        if (accessManager == address(0)) {
+            revert Errors.GroupingModule__ZeroAccessManager();
+        }
+        __ReentrancyGuard_init();
+        __UUPSUpgradeable_init();
+        __ProtocolPausable_init(accessManager);
+    }
+
+    /// @notice Registers a Group IPA.
+    /// @param groupPool The address of the group pool.
+    /// @return groupId The address of the newly registered Group IPA.
+    function registerGroup(address groupPool) external whenNotPaused returns (address groupId) {
```

Comment:
may require nonReentrant modifier as we are sending the NFT to the msg.sender

Reply:
It should be ok, as the group NFT is developed by us.

Reply:
offline discussed, it would safer to add nonReentrant guard as the nft receiver might implement onReceived function.

Added the nonReentrant then.

Reply:
Is this change committed?